rust/scx_utils/src/topology.rs

```diff
@@ -449,10 +487,24 @@ fn create_insert_cpu(cpu_id: usize, node: &mut Node, online_mask: &Cpumask) -> R
         span: Cpumask::new()?,
     });
 
+    let core_type = match avg_cpu_freq {
+        Some(avg_freq) => {
+            if max_freq > avg_freq {
```

Comment:
The kernel's capacity numbers are nice because it gives the relative max performance. `driver/base/arch_topology.c` exposes CPU capacity but it's not implemented for x86, unfortunately. `scx_bpf_cpuperf_cap()` exposes this but that's not accessible without running BPF progs. Given that `scx_bpf_cpuperf_cap()` can be called from any BPF progs, I wonder whether it'd be possible to make topology crate access those numbers? What do you think?

Reply:
I was thinking of having a version available on for userspace via the `Topology` and a separate bpf helpers in `scheds/include/scx/topology.bpf.h` or something like that. I had also tried `scx_bpf_cpuperf_cap` in the past and as you mention it doesn't work on x86.

Reply:
Ah, right. This is broken on x86. Once this is fixed, what I'm saying is that it might be useful to make `Topology` crate to query `scx_bpf_cpuperf_cap()` using its own embedded BPF program and then populate cap values accordingly in the struct so that userspace can use them.

Reply:
I also tried `scx_bpf_cpuperf_cap()`, but it returns the same 1024 for all CPU types.